rest-service/manager_rest/deployment_update/manager.py

```diff
@@ -88,58 +87,30 @@ def stage_deployment_update(self,
         # enables reverting to original blueprint resources
         deployment = self.sm.get(models.Deployment, deployment_id)
         old_blueprint = deployment.blueprint
-        file_server_root = config.instance.file_server_root
-        blueprint_resource_dir = os.path.join(file_server_root,
-                                              'blueprints',
-                                              old_blueprint.tenant_name,
-                                              old_blueprint.id)
-        runtime_only_evaluation = runtime_only_evaluation or \
-            deployment.runtime_only_evaluation
-        # The dsl parser expects a URL
-        blueprint_resource_dir_url = 'file:{0}'.format(blueprint_resource_dir)
-        app_path = os.path.join(file_server_root, app_dir, app_blueprint)
-
-        # parsing the blueprint from here
-        try:
-            plan = tasks.parse_dsl(
-                app_path,
-                resources_base_path=file_server_root,
-                additional_resources=[blueprint_resource_dir_url],
-                **app_context.get_parser_context()
-            )
-        except parser_exceptions.DSLParsingException as ex:
-            raise manager_exceptions.InvalidBlueprintError(
-                'Invalid blueprint - {0}'.format(ex))
+        runtime_only_evaluation = (runtime_only_evaluation or
+                                   deployment.runtime_only_evaluation)
+        parsed_deployment = get_parsed_deployment(old_blueprint,
+                                                  app_dir,
+                                                  app_blueprint)
 
         # Updating the new inputs with the deployment inputs
         # (overriding old values and adding new ones)
         old_inputs = copy.deepcopy(deployment.inputs)
         new_inputs = {k: old_inputs[k]
-                      for k in plan.inputs.keys() if k in old_inputs}
+                      for k in parsed_deployment.inputs.keys()
```

Comment:
i usually prefer to drop `.keys()` from dict iteration calls, because iterating over the dict itself gives you keys, so no real need to create an additional object (and in py2, it'll be a full list of keys - so if the dict was a million entries, you're now creating an additional list of a million items as well)

Reply:
Great call, deleted